pkg/controller/controller.go

```diff
@@ -535,6 +410,37 @@ func (c *Controller) updateImageAnnotationInfo(imageSha string, newInfo string)
 	return true
 }
 
+func (c *Controller) mountAndScan(id string, imageRef string, imageSha string) (report string, err error) {
+
+	scanDirectory := "/data/scanDir/scratch"
+	//cleanup first
+	os.RemoveAll(scanDirectory)
+	os.MkdirAll(scanDirectory, os.ModePerm)
+
+	scanOptions := container.NewDefaultImageMounterOptions()
+	scanOptions.DstPath = scanDirectory
+	scanOptions.Image = imageSha
+	mounter := container.NewDefaultImageMounter(*scanOptions)
+	_, image, _ := mounter.Mount()
+	rhaiDir := path.Join(scanOptions.DstPath, "etc", "redhat-access-insights")
+	machineidPath := path.Join(rhaiDir, "machine-id")
+	os.MkdirAll(rhaiDir, os.ModePerm)
+	err = ioutil.WriteFile(machineidPath, []byte("deeznuts"), 0644)
```

Comment:
don't need to spoof machine-id anymore. see this commit

https://github.com/RedHatInsights/insights-ocp-scanner/pull/4/commits/4ed41b9f7b66a5b58cb56d8d51475c514bad3c29

Reply:
otherwise the rest looks good